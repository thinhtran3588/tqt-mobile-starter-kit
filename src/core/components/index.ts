export * from '@tqt/mobile/lib/components';
export * from './icon/icon.component';
export {
  Provider as PaperProvider,
  DefaultTheme as PaperDefaultTheme,
  Colors,
  Button,
  Appbar,
  Card,
  ActivityIndicator,
  Avatar,
  Badge,
} from 'react-native-paper';
export {StatusBar, Alert, ScrollView} from 'react-native';
