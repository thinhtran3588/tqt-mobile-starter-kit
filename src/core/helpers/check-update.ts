import {Platform, Alert} from 'react-native';
import codepush from 'react-native-code-push';
import {config} from '@core/config';

export const checkUpdate = async (isTester: boolean): Promise<void> => {
  let deploymentKey = '';
  if (Platform.OS === 'android') {
    deploymentKey = isTester ? config().android.codepush.stagingKey : config().android.codepush.productionKey;
  } else {
    deploymentKey = isTester ? config().ios.codepush.stagingKey : config().ios.codepush.productionKey;
  }
  try {
    await codepush.sync({deploymentKey, installMode: codepush.InstallMode.ON_NEXT_RESTART}, (status) => {
      if (status === codepush.SyncStatus.UPDATE_INSTALLED) {
        Alert.alert('Alert', 'A new version is installed. Restart now?', [
          {
            text: 'Restart',
            onPress: () => codepush.restartApp(),
          },
          {
            text: 'Close',
            style: 'cancel',
          },
        ]);
      }
    });
  } catch (error) {
    // ignore
  }
};
