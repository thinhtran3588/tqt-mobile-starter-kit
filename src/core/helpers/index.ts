import merge from 'lodash/merge';

export * from './sleep';
export {merge};
