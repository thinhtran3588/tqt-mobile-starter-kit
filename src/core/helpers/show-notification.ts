import Toast, {ToastOptions} from 'react-native-root-toast';
import {COLORS_LOOKUP, DARK_BACKGROUND_COLOR, LIGHT_BACKGROUND_COLOR} from '../contexts';

type NotificationType = 'SUCCESS' | 'WARNING' | 'ERROR' | 'INFO';

export interface NotificationParams extends ToastOptions {
  message: string;
  type?: NotificationType;
}

export interface NotificationTheme {
  theme: 'dark' | 'light';
  warning: string;
  error: string;
  success: string;
}

interface LastMessage {
  message: string;
  timestamp: number;
  type: NotificationType;
}

let notificationTheme: NotificationTheme = {
  theme: 'light',
  warning: COLORS_LOOKUP.ORANGE.color,
  error: COLORS_LOOKUP.RED.color,
  success: COLORS_LOOKUP.GREEN.color,
};

export const setNotificationTheme = (theme: 'dark' | 'light'): void => {
  if (theme === 'light') {
    notificationTheme = {
      theme: 'light',
      warning: COLORS_LOOKUP.ORANGE.color,
      error: COLORS_LOOKUP.RED.color,
      success: COLORS_LOOKUP.GREEN.color,
    };
  } else {
    notificationTheme = {
      theme: 'light',
      warning: COLORS_LOOKUP.ORANGE.darkColor,
      error: COLORS_LOOKUP.RED.darkColor,
      success: COLORS_LOOKUP.GREEN.darkColor,
    };
  }
};

export const showNotification = (params: NotificationParams): void => {
  const {
    message,
    type = 'INFO',
    duration = Toast.durations.SHORT,
    position = Toast.positions.TOP,
    delay = 0,
    ...other
  } = params;

  let backgroundColor: string = notificationTheme.theme === 'dark' ? LIGHT_BACKGROUND_COLOR : DARK_BACKGROUND_COLOR;
  const textColor: string = LIGHT_BACKGROUND_COLOR;
  if (type === 'SUCCESS') {
    backgroundColor = notificationTheme.success;
  } else if (type === 'WARNING') {
    backgroundColor = notificationTheme.warning;
  } else if (type === 'ERROR') {
    backgroundColor = notificationTheme.error;
  }

  Toast.show(message, {
    ...other,
    duration,
    position,
    delay,
    backgroundColor,
    textColor,
    containerStyle: {
      top: 30,
      borderRadius: 20,
    },
    opacity: 0.9,
  });
};
