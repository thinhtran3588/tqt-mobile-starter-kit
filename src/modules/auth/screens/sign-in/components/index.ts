export * from './social-sign-in/social-sign-in.component';
