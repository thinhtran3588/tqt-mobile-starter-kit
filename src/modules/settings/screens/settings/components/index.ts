export * from './theme-setting/theme-setting.component';
export * from './primary-color-setting/primary-color-setting.component';
export * from './general-info/general-info.component';
export * from './language-setting/language-setting.component';
