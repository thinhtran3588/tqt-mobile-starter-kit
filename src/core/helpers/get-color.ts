import {ThemeState} from '@settings/models';
import {ColorType} from '@app/core';

export const getColor = (type: ColorType = 'SUCCESS', theme: ThemeState): string => {
  let color: string;
  switch (type) {
    case 'ERROR':
      color = theme.colors.error;
      break;
    case 'WARNING':
      color = theme.colors.warning;
      break;
    case 'INFO':
      color = theme.colors.info;
      break;
    case 'PRIMARY':
      color = theme.colors.primary;
      break;
    default:
      color = theme.colors.success;
  }
  return color;
};
