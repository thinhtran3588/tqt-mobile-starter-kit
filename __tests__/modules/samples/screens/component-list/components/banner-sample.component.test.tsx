import React from 'react';
import {render, fireEvent, toJSON} from '@test-utils';
import {sleep} from '@core/helpers';
import {BannerSample} from '@samples/screens/component-list/components';

it('presses Close Banner button', async () => {
  const {baseElement, getByText} = render(<BannerSample />);

  expect(toJSON(baseElement)).toMatchSnapshot();
  fireEvent.press(getByText('Close Banner'));
  await sleep(1000);
  // render updated view
  expect(toJSON(baseElement)).toMatchSnapshot();
});
