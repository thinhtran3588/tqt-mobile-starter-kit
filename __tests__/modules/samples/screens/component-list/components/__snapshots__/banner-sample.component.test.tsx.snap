// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`presses Close Banner button 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "elevation": 1,
            "marginBottom": 5,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 0.75,
          }
        }
        testID="banner"
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "maxWidth": 960,
                "overflow": "hidden",
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "height": 0,
              }
            }
          />
          <View
            style={Object {}}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginBottom": 0,
                  "marginHorizontal": 8,
                  "marginTop": 16,
                }
              }
            >
              <View
                style={
                  Object {
                    "margin": 8,
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 40,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰓎
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                    },
                    Object {
                      "flex": 1,
                      "margin": 8,
                    },
                  ]
                }
              >
                Samples of all components
                <Button
                  title="Close"
                />
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "margin": 4,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`presses Close Banner button 2`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "elevation": 1,
            "marginBottom": 5,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 0.75,
          }
        }
        testID="banner"
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "maxWidth": 960,
                "overflow": "hidden",
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "height": 0,
              }
            }
          />
          <View
            style={
              Object {
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transform": Array [
                  Object {
                    "translateY": -0,
                  },
                ],
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginBottom": 0,
                  "marginHorizontal": 8,
                  "marginTop": 16,
                }
              }
            >
              <View
                style={
                  Object {
                    "margin": 8,
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 40,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰓎
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                    },
                    Object {
                      "flex": 1,
                      "margin": 8,
                    },
                  ]
                }
              >
                Samples of all components
                <Button
                  title="Close"
                />
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "margin": 4,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
